docs/errors/E0383.md

```diff
@@ -0,0 +1,15 @@
+# E0383: variable assignment to self is no-op
+
+The following is a no-op and will warn:
+
+```javascript
+x = x;
+```
+
+The following is not necessarily a no-op and will not warn:
+
+```javascript
+i.x = i.x;
+```
```

Comment:
I think this example is confusing. It has nothing to do with the diagnostic or how to fix the diagnostic.

In our docs, the second code example shows what a fixed program looks like. I'd replace this example with what the user should write instead of the first example. In this case, though, that would be an empty code example.

Perhaps the following is a better example:
```javascript
let next = 0;
if (previous !== null) {
  next = next;
}
switchToPage(next);
```
```javascript
let next = 0;
if (previous !== null) {
  next = previous;
}
switchToPage(next);
```